samples/subsys/fs/fat_fs/prj_stm32f746g_disco.conf

```diff
@@ -0,0 +1,5 @@
+CONFIG_DISK_ACCESS=y
+CONFIG_LOG=y
+CONFIG_FILE_SYSTEM=y
+CONFIG_FAT_FILESYSTEM_ELM=y
+CONFIG_PRINTK=y
```

Comment:
IIC These are all already available in prj.conf, so this file should'nt be needed

Reply:
The default prj.conf for that sample has:
`CONFIG_DISK_ACCESS_SPI_SDHC=y`
Which causes a build failure because the stm32 boards don't have that.

Reply:
As I commented in #24787, CONFIG_DISK_ACCESS_SPI_SDHC=y might be better put in each boards .conf so you can have this board to be treated like other boards under /boards

Reply:
@anthonybrandon https://github.com/zephyrproject-rtos/zephyr/pull/24787 is now taking this into account.
You just need to rebase and use same config scheme

Reply:
okay, done.

Reply:
@anthonybrandon This is not exactly what I expect. Sorry if not clear. The following are already available in prj.cnf:
```
CONFIG_DISK_ACCESS=y
CONFIG_LOG=y
CONFIG_FILE_SYSTEM=y
CONFIG_FAT_FILESYSTEM_ELM=y
```
So, if you really need printk, you just need to add a file `samples/subsys/fs/fat_fs/boards/stm32f746g_disco.conf` with the following:
```
CONFIG_PRINTK=y
```